cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CapacityConfiguration.java

```diff
@@ -0,0 +1,257 @@
+    }
+
+    private static Map<Integer, BrokerCapacityOverride> processBrokerCapacityOverrides(Reconciliation reconciliation,
+                                                                                       Set<NodeRef> kafkaBrokerNodes,
+                                                                                       BrokerCapacity brokerCapacity) {
+        Map<Integer, BrokerCapacityOverride> overrideMap = new HashMap<>();
+
+        if (brokerCapacity != null && brokerCapacity.getOverrides() != null && !brokerCapacity.getOverrides().isEmpty()) {
+            for (BrokerCapacityOverride override : brokerCapacity.getOverrides()) {
+                List<Integer> ids = override.getBrokers();
+
+                for (int id : ids) {
+                    if (overrideMap.containsKey(id)) {
+                        LOGGER.warnCr(reconciliation, "Duplicate broker id {} found in overrides, using first occurrence.", id);
+                    } else if (kafkaBrokerNodes.stream().noneMatch(node -> node.nodeId() == id)) {
+                        LOGGER.warnCr(reconciliation, "Ignoring broker capacity override for unknown node ID {}", id);
+                    } else {
+                        overrideMap.put(id, override);
+                    }
+                }
+            }
+        }
+
+        return overrideMap;
+    }
+
+    private static TreeMap<Integer, CapacityEntry> generateCapacityEntries(Reconciliation reconciliation,
+                                         CruiseControlSpec spec,
+                                         Set<NodeRef> kafkaBrokerNodes,
+                                         Map<String, Storage> kafkaStorage,
+                                         Map<String, ResourceRequirements> kafkaBrokerResources) {
+        TreeMap<Integer, CapacityEntry> capacityEntries = new TreeMap<>();
+        BrokerCapacity generalBrokerCapacity = spec.getBrokerCapacity();
+        Map<Integer, BrokerCapacityOverride> brokerCapacityOverrideMap = processBrokerCapacityOverrides(reconciliation,
+                kafkaBrokerNodes, generalBrokerCapacity);
+
+        for (NodeRef node : kafkaBrokerNodes) {
+            BrokerCapacityOverride brokerCapacityOverride = brokerCapacityOverrideMap.get(node.nodeId());
+
+            DiskCapacity disk = new DiskCapacity(kafkaStorage.get(node.poolName()), node.nodeId());
+            CpuCapacity cpu = new CpuCapacity(generalBrokerCapacity, brokerCapacityOverride, kafkaBrokerResources.get(node.poolName()));
+            InboundNetworkCapacity inboundNetwork = new InboundNetworkCapacity(generalBrokerCapacity, brokerCapacityOverride);
+            OutboundNetworkCapacity outboundNetwork = new OutboundNetworkCapacity(generalBrokerCapacity, brokerCapacityOverride);
+
+            CapacityEntry capacityEntry = new CapacityEntry(node.nodeId(), disk, cpu, inboundNetwork, outboundNetwork);
+            capacityEntries.put(node.nodeId(), capacityEntry);
```

Comment:
Maybe we can just drop the variable creation and doing it when putting the object into the map.

Reply:
Fixed.